version: '3'
services:
  kafka:
    container_name: kafka_demo    
    build: 
      context: ./KAFKA_MQ/
    ports:
      - "9092:9092"
      - "9092:9092/udp"
    networks: 
      - custom-network
  api_net:
    container_name: kafka_api
    build: 
      context: ./KAFKA_API/
    ports: 
      - "8088:80"
    networks: 
      - custom-network      
    depends_on: 
      - kafka
  rabbitmq:
    build:
        context: ./RABBIT/
        dockerfile: Dockerfile
    container_name: rabbitMQ-lab
    restart: always
    ports:
        - "15672:15672"
        - "5672:5672"
    hostname: rabbit_host
    environment: 
        RABBITMQ_DEFAULT_USER: test
        RABBITMQ_DEFAULT_PASS: test
        RABBITMQ_VM_MEMORY_WATERMARK: 0.5
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:15672"]
      interval: 30s
      timeout: 10s
      retries: 5        
    networks:
        - rabbit_network
  rabbitmq_consumer:
    build:
        context: ./RABBIT/
        dockerfile: node.Dockerfile
    container_name: rabbitMQ-consumer
    restart: on-failure
    depends_on: 
      - rabbitmq  
    links: 
      - rabbitmq           
    networks:
        - rabbit_network                
networks:
  custom-network:
    driver: bridge
  rabbit_network: 
    driver: bridge 